src/main/scala/com/socrata/regioncoder/Main.scala

```diff
@@ -0,0 +1,98 @@
+import com.socrata.http.server.{HttpRequest, HttpResponse, SocrataServerJetty}
+import com.socrata.http.server.responses._
+import com.socrata.http.server.implicits._
+import com.socrata.http.server.util.RequestId
+import com.socrata.http.common.AuxiliaryData
+import com.socrata.regioncoder._
+import com.socrata.regioncoder.config.RegionCoderConfig
+import com.socrata.soda.external.SodaFountainClient
+import com.socrata.curator.ServerProvider.RetryOnAllExceptionsDuringInitialRequest
+import com.socrata.curator.{CuratorFromConfig, DiscoveryFromConfig}
+import com.socrata.http.server.curator.CuratorBroker
+import com.typesafe.config.ConfigFactory
+import com.rojoma.simplearm.v2._
+import org.slf4j.{MDC, LoggerFactory}
+
+class Main(config: RegionCoderConfig) {
+  val log = LoggerFactory.getLogger(classOf[Main])
+
+  implicit def executorResource = Resource.executorShutdownNoTimeout
+
+  def exceptionalResult(e: Exception): HttpResponse = {
+    log.error("Request errored out", e)
+    InternalServerError ~> Content("text/plain", s"${e.getClass.getSimpleName}: ${e.getMessage}\n${e.getStackTraceString}\n")
+  }
+
+  def runServer() {
+    for {
+      httpExecutor <- managed(Executors.newCachedThreadPool())
+      httpClient <- managed(new HttpClientHttpClient(
+                              httpExecutor,
+                              HttpClientHttpClient.defaultOptions.
+                                withLivenessChecker(NoopLivenessChecker).
+                                withUserAgent("region-coder")))
+      curator <- CuratorFromConfig(config.curator)
+      discovery <- DiscoveryFromConfig(classOf[AuxiliaryData], curator, config.discovery)
+      sodaFountain <- managed(new SodaFountainClient(
+                                httpClient,
+                                discovery,
+                                config.sodaFountain.serviceName,
+                                config.curator.connectTimeout,
+                                config.sodaFountain.maxRetries,
+                                RetryOnAllExceptionsDuringInitialRequest,
+                                throw ServiceDiscoveryException("No Soda Fountain servers found")) {
+                              }).and(_.start())
+    } {
+      val broker = new CuratorBroker(discovery, config.discovery.address, config.discovery.name, None)
+
+      val baseHandler = new RegionCoderServlet(config, sodaFountain)
+
+      val wrappedHandler = { (req: HttpRequest) =>
+        MDC.clear()
+        MDC.put(RequestId.ReqIdHeader, req.requestId)
+
+        log.info("{} - {}{}", req.method, req.requestPathStr, req.queryStr.fold("")("?" + _))
+
+        val result =
+          try {
+            baseHandler.handle(req)
+          } catch {
+            case e: Exception =>
+              exceptionalResult(e)
+          }
+
+        { (resp: HttpServletResponse) =>
+          try {
+            result(resp)
+          } catch{
+            case e: Exception =>
```

Comment:
I was noticing this in some other NBE services-- it seems to me like the change you made to the SocrataServerJetty already does this handling? Am I misunderstanding?

https://github.com/socrata-platform/socrata-http/blob/master/socrata-http-jetty/src/main/scala/com/socrata/http/server/Handlers.scala#L38

Reply:
It does.  This is here for the uniform logging.

Reply:
What do you mean? This part should never run since it's already caught, no?

Reply:
No, this function here is `handler` over there, so this is inside that try.

Reply:
right, I see, sorry D: